File: evento-app/evento-app/src/app/sistema/cliente/Cliente.ts
import { Vendedor } from "../vendedores/Vendedor";

export class Cliente{
    id: number;
    nome: string;
    email: string;
    razao: string;
    telefone: string;
    idVendedor: number;
    nomeVendedor: string;
    cnpj: string;
    dataCadastro: string;
    aceitaReceberEmails: boolean;

    constructor(
        id: number,
        nome: string,
        email: string,
        razao: string,
        telefone: string,
        idVendedor: number,
        nomeVendedor: string,
        cnpj: string,
        dataCadastro: string,
        aceitaReceberEmails: boolean,){
        this.id = id;
            this.nome = nome;
            this.email = email;
            this.razao = razao;
            this.telefone = telefone;
            this.idVendedor = idVendedor;
            this.nomeVendedor = nomeVendedor;
            this.cnpj = cnpj;
            this.dataCadastro = dataCadastro;
            this.aceitaReceberEmails = aceitaReceberEmails
        }
}

export class ClienteDTO{
    id: number;
    nome: string;
    razao: string;
    telefone: string;
    vendedor: Vendedor;
    cnpj: string;
    dataCadastro: string;
    eventoId: any[];
    aceitaReceberEmails: boolean;

    constructor(    id: number,
        nome: string,
        razao: string,
        telefone: string,
        vendedor: Vendedor,
        cnpj: string,
        dataCadastro: string,
        eventoId: any[],
        aceitaReceberEmails: boolean,){
            this.id = id;
            this.nome = nome;
            this.razao = razao;
            this.telefone = telefone;
            this.vendedor = vendedor;
            this.cnpj = cnpj;
            this.dataCadastro = dataCadastro;
            this.eventoId = eventoId;
            this.aceitaReceberEmails = aceitaReceberEmails
        }
}

File: evento/src/main/java/com/myevent/eventos/entity/Cliente.java
package com.myevent.eventos.entity;

import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;
    private String razao;
    private String telefone;
    private String email;
    private String cnpj;
    private String data;
    private Boolean aceitaReceberEmails;

    @ManyToOne
    private Vendedor vendedor;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "evento_cliente",
        joinColumns = @JoinColumn(name = "cliente_id"),
        inverseJoinColumns = @JoinColumn(name = "evento_id"))
    @JsonIgnore
    private List<Evento> eventos = new ArrayList<>();
    
    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "evento_interesse_cliente",
        joinColumns = @JoinColumn(name = "cliente_id"),
        inverseJoinColumns = @JoinColumn(name = "evento_interesse_id"))
    @JsonIgnore
    private List<Evento> eventosInteresse = new ArrayList<>();

   
}
